ClamavPlugin.inc.php

```diff
@@ -181,18 +181,37 @@ function getClamVersion($path = '', $type=self::TYPE_EXECUTABLE) {
 
 	/**
 	 * Private helper method to scan a file using the clamscan executable, returning a virus message if matched
-	 * @param $uploadedFileField string The field index in $_FILES[] which references the uploaded file
+	 * @param $props array Information about the uploaded file from the OJS database
+	 * @param $uploadedFile string Path to the uploaded file in OJS's files directory
 	 * @return string
 	 */
-	function _clamscanFile($uploadedFileField) {
-		if ($this->getClamVersion() && !empty($uploadedFileField) && isset($_FILES[$uploadedFileField]['tmp_name'])) {
+	function _clamscanFile($props, $uploadedFile) {
+		if ($this->getClamVersion() && !empty($uploadedFile)) {
 			$output = "";
 			$exitCode = "";
-			$uploadedFile = $_FILES[$uploadedFileField]['tmp_name'];
-			$scan = exec($this->getSetting(CONTEXT_SITE, 'clamavPath') . ' -i --no-summary ' . $uploadedFile, $output, $exitCode);
-			// If the scan returned anything, remove the temporary filename and report the error
+			$scan = exec($this->getSetting(CONTEXT_SITE, 'clamavPath') . ' --no-summary -i ' . $uploadedFile, $output, $exitCode);
+			// Virus detected
 			if ($exitCode === 1) {
-				return $this->_clamscanBuildAlert($uploadedFileField, 'plugins.generic.clamav.uploadBlocked', str_replace($uploadedFile . ': ', '', $scan));
+				//For Clam's error message: pull out just the part after the colon
+				preg_match('/:(.*)/',$output[0],$matches);
+				$virusID=trim($matches[1]);
+				$schemaService = Services::get('schema');
+				$threatName=["threatname"=>$virusID];
+
+				import('lib.pkp.classes.validation.ValidatorFactory');
+				$validator = \ValidatorFactory::make(
+					$props,
+					$schemaService->getValidationRules(SCHEMA_SUBMISSION_FILE, $allowedLocales), 
```

Comment:
`$allowedLocales` appears to be undefined.

Reply:
I can get this parameter from info passed to clamScanHandleUpload, but I am not clear on what value it provides.

Reply:
The functionality we are hooking here is generally performing validation of form-based user input against defined schemas.  Locale checks probably ensure that if the form only supports three languages, only those three languages are accepted (and perhaps content is defaulted in missing languages).

For your purpose, what you want to know is whether it makes sense (and is worthwhile) to utilize formal validation errors through the `ValidatorFactory`, or if the better approach is just to render the formatted errors directly to pass back to the caller.  Concretely, is it legitimate to create a new "field" with the name 'virusDetected' to apply our error message to, when no such field exists in the schema?  Should we be using the `SchemaService` to generate validation rules in the `ValidatorFactory`, or should we generate validation rules another way?

Note that [`ValidatorFactory` wraps Laravel functionality](https://github.com/pkp/pkp-lib/blob/149c991ad9dc6d7a067593d1a1d163b7190f0824/classes/validation/ValidatorFactory.inc.php#L24-L31), so the real work of creating and using validators is documented here: https://laravel.com/docs/9.x/validation

I recommend reading a bit and then writing up a question to the core devs in Slack to get clarity on what the best approach would be here.